model/channel.go

```diff
@@ -281,9 +284,11 @@ func (o *Channel) IsValid() *AppError {
 		return NewAppError("Channel.IsValid", "model.channel.is_valid.creator_id.app_error", nil, "", http.StatusBadRequest)
 	}
 
-	userIds := strings.Split(o.Name, "__")
-	if o.Type != ChannelTypeDirect && len(userIds) == 2 && IsValidId(userIds[0]) && IsValidId(userIds[1]) {
-		return NewAppError("Channel.IsValid", "model.channel.is_valid.name.app_error", nil, "", http.StatusBadRequest)
+	if o.Type != ChannelTypeDirect && o.Type != ChannelTypeGroup {
+		userIds := strings.Split(o.Name, "__")
+		if ok := gmNameRegex.MatchString(o.Name); ok || o.Type != ChannelTypeDirect && len(userIds) == 2 && IsValidId(userIds[0]) && IsValidId(userIds[1]) {
```

Comment:
`o.Type != ChannelTypeDirect && len(userIds) == 2 && IsValidId(userIds[0]) && IsValidId(userIds[1])`

I know this part of the condition wasn't added in this PR, but it looks like its not covered in the tests. Could we add a test for this condition as well, to lock in the behaviour? It should only be two more lines:

```
o.Name = "71b03afcbb2d503d49f87f057549c43db4e19f92__pazfdvmko08qhzd0icg76zo9gr3w42ah05"
require.NotNil(t, o.IsValid())
```

Reply:
Good call 👍